src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/BKTree.cs

```diff
@@ -70,25 +70,20 @@ public void Find(ref TemporaryArray<string> result, string value, int? threshold
         if (_nodes.Length == 0)
             return;
 
-        var lowerCaseCharacters = ArrayPool<char>.GetArray(value.Length);
-        try
-        {
-            for (var i = 0; i < value.Length; i++)
-                lowerCaseCharacters[i] = CaseInsensitiveComparison.ToLower(value[i]);
+        Span<char> lowerCaseCharacters = value.Length < 512
+            ? stackalloc char[value.Length]
+            : new char[value.Length];
```

Comment:
I would still use the pool for large strings.

Reply:
Just here, or in EditDistance too? I'm not sure about here, but I wouldn't think the EditDistance code would almost ever exceed 512 chars.

Reply:
Just to make sure, I don't think the old code was pooling large strings, as ArrayPool checked against MaxPooledArraySize for both retrieving and returning from the pool.

Reply:
Ah. Never mind :-)